crates/sui-core/src/epoch/randomness.rs

```diff
@@ -546,22 +456,30 @@ impl Inner {
             warn!("ignoring equivocating DKG Message from authority {authority:?} pretending to be PartyId {party_id:?}");
             return Ok(());
         }
-        match self.party.process_message(msg, &mut rand::thread_rng()) {
-            Ok(processed) => {
-                self.processed_messages
-                    .insert(processed.message.sender, processed.clone());
-                batch.insert_batch(
-                    &self.tables()?.dkg_processed_messages,
-                    std::iter::once((processed.message.sender, processed)),
-                )?;
-            }
-            Err(err) => {
-                debug!("random beacon: error while processing DKG Message: {err:?}");
-            }
+        if self.enqueued_messages.contains_key(&msg.sender)
+            || self.processed_messages.contains_key(&msg.sender)
+        {
+            info!("ignoring duplicate DKG Message from authority {authority:?}");
+            return Ok(());
         }
+
+        let party = self.party.clone();
+        self.enqueued_messages.insert(
+            msg.sender,
+            tokio::task::spawn_blocking(move || {
```

Comment:
can we add a TODO here to reflect the discussion about potentially not eagerly processing all messages?

Reply:
done